### PR TITLE
auth: unit-test for authStore.AuthDisable()

### PR DESCRIPTION
**Pull request by Rushit**
This will cover unit-test for AuthDisable in store.go

/cc @mitake @xiang90

### CODE REVIEW

File: auth/store_test.go [hunk 1]
```diff
@@ -448,6 +448,26 @@ func TestAuthInfoFromCtx(t *testing.T) {
...
+
+	as.AuthDisable()
+
+	ctx := context.WithValue(context.WithValue(context.TODO(), "index", uint64(2)), "simpleToken", "dummy")
+	_, err := as.Authenticate(ctx, "foo", "bar")
+	if err != ErrAuthNotEnabled {
+		t.Errorf("expected %v, got %v", ErrAuthNotEnabled, err)
+	}
+
+	// Disabling disabled auth again and trying to Authenticate
```
Comment (xiang90) [inline auth/store_test.go]: add a comment about why we want to try disabling auth again?

**Comment by xiang90** [comment]
LGTM. Defer to @mitake

File: auth/store_test.go (hunk 1 above)
Comment (Rushit) [inline auth/store_test.go]: Disabling disabled auth to make sure it can return safely if store is already disabled.

Comment (xiang90) [inline auth/store_test.go]: @Rushit Yea. Probably add the comment into the code as a comment? I know why we need to test it. But it just reads strange without an explanation.

Comment (xiang90) [inline auth/store_test.go]: @Rushit Great! thank you!

Comment (Rushit) [inline auth/store_test.go]: done.

**Comment by mitake** [comment]
lgtm, thanks @Rushit 

@xiang90 BTW, can I ignore jenkins-proxy-ci? I couldn't see something is going under this PR (the link of Detailed returned 404)

**Comment by xiang90** [comment]
> can I ignore jenkins-proxy-ci? 

ignore it for now.

**Pull request merged**